### ISSUE TITLE
Details on KTH and BAIR Validation Sets

### ISSUE THREAD

**Issue by hanghang177** [kaanakan/slamp#1]
Hi! Thanks for providing the implementation of SLAMP. In the data processing scripts (`data/kth.py` and `data/bair.py`), how do you generate `kth_valset_40.npz` and `bair_valset_30.npz`? Is it following the SRVP's code for generating test sets? Could you please provide some details on those sets? Thank you!

**Comment by kaanakan** [comment]
Hi, thank you for your interest in our work!

For generating test sets, we followed [SRVP](https://github.com/edouardelasalles/srvp)'s code. For generating KTH test set and BAIR test set, you can follow these links ([KTH](https://github.com/edouardelasalles/srvp#kth), [BAIR](https://github.com/edouardelasalles/srvp#bair)).

If you have further questions, please feel free to ask.

**Comment by hanghang177** [comment]
Hi, thanks a lot for your response! However, even though your code is mostly following SRVP's code, the generation of the validation sets is different.

In particular, SRVP uses a random 95/5 train/val split by using a fixed seed 42, whereas your data loader utilizes the same random 95% training set but a separate npz file for the validation set. SRVP's code doesn't generate validation set as npz file. For example, for BAIR, your data processing code does the following for the validation set. The SRVP preprocessing code does not generate `bair_valset_30.npz`.

`sequences = np.load(join(data_dir, 'processed_data', 'bair_valset_30.npz'))['sequences']`

I'm really interested in how you generate the validation set. Is it the same as the 5% selected by random seed 42 as in SRVP? Or is it something different? Also, would you mind providing the data preprocessing code for KTH, BAIR, KITTI, and Cityscapes as there are clearly differences from the SRVP repo?

Again, thank you so much for the help!

**Comment by kaanakan** [comment]
Hi,

We thought that validation takes a lot of time, and thus, we decided to create a held-out validation set for both datasets. If you check SRVP's code, you will see they are validating their model for a fixed number of iterations (25 by default, `n_iter_test` argument in args.py in SRVP's code). However, our code had to go over the whole validation set. Therefore, we decided to do that trick. 

For the test sets, we are using SVG's test sets as it is used in SRVP's code as well.

You can find the validation sets for KTH and BAIR datasets on [this link](https://drive.google.com/drive/folders/1Mvs5gFCyEI8XluKkrkn7B8XDlNn0iu7K?usp=sharing). 

If you have further questions, please feel free to ask. 
We are happy to help.

**Comment by hanghang177** [comment]
Thank you so much for providing the two files! Would you mind telling me how you generate the two validation sets (like which image sequences or person ids)?
